src/storage/src/hummock/sstable/forward_sstable_iterator.rs

```diff
@@ -70,13 +91,15 @@ impl SstableIterator {
             options,
             preload_end_block_idx: 0,
             preload_retry_times: 0,
+            read_block_meta_range: (start_idx, end_idx),
         }
     }
 
     fn init_block_prefetch_range(&mut self, start_idx: usize) {
         self.preload_end_block_idx = 0;
         if let Some(bound) = self.options.must_iterated_end_user_key.as_ref() {
-            let block_metas = &self.sst.meta.block_metas;
+            let block_metas = &self.sst.meta.block_metas
+                [self.read_block_meta_range.0..=self.read_block_meta_range.1];
             let next_to_start_idx = start_idx + 1;
```

Comment:
hmm... I think the `idx` here can be completely wrong given that `block_metas` are sliced in L101.

Reply:
Thanks, I happened to find the bug and have fixed it in a new commit .